Modules/zipimport.c

```diff
@@ -1497,21 +1497,86 @@ get_mtime_of_source(ZipImporter *self, PyObject *path)
     return mtime;
 }
 
+static PyObject *
+normalize_zip_path(PyObject *name)
+{
+    PyObject *sep1 = NULL, *sep2 = NULL, *nameobj = NULL;
+    sep1 = PyUnicode_FromFormat("%c", (int)'/');
+    if (sep1 == NULL) {
+        goto exit;
+    }
+    sep2 = PyUnicode_FromFormat("%c", SEP);
+    if (sep2 == NULL) {
+        goto exit;
+    }
+    nameobj = PyUnicode_Replace(name, sep1, sep2, -1);
+exit:
+    Py_XDECREF(sep1);
+    Py_XDECREF(sep2);
+    return nameobj;
+}
+
+/* Select code origin for unmarshalled code.
+   Return borrowed reference or NULL if lookup failed. */
+static PyObject *
+select_code_origin(ZipImporter *self, PyCodeObject *code, PyObject *toc_entry)
+{
+    PyObject *nameobj, *toc_origin;
+    nameobj = normalize_zip_path(code->co_filename);
+    if (nameobj == NULL) {
+        return NULL;
+    }
+    toc_origin = PyDict_GetItem(self->files, nameobj);
+    Py_DECREF(nameobj);
+    if (toc_origin == NULL) {
+        return PyTuple_GetItem(toc_entry, 0);
+    }
+    return PyTuple_GetItem(toc_origin, 0);
+}
+
+static void
+overwrite_code_origin(PyCodeObject *code, PyObject *code_origin)
+{
+    Py_ssize_t pos, len;
+    PyObject *item;
+
+    Py_INCREF(code_origin);
+    Py_DECREF(code->co_filename);
+    code->co_filename = code_origin;
+
+    if (PyTuple_Check(code->co_consts)) {
+        len = PyTuple_Size(code->co_consts);
+        for (pos = 0; pos < len; ++pos) {
+            item = PyTuple_GET_ITEM(code->co_consts, pos);
```

Comment:
Here PyTuple_GetItem replaced with PyTuple_GET_ITEM, since PyTuple_Check is checked above and pos is in range of valid tuple indices.